### PR TITLE
provide several options to have more control over the provisioner

### PR DESCRIPTION
**Pull request by Itxaka**
Provides 4 new options to have more control over the provisioner:

chef_binary: Allows to provide a path for the chef-{client, solo, zero} binary to allow
control over what binary to launch
chef_options: Allows to pass extra options to the chef binary call or avoid the default -z option
chef_log_level: Allows to pass the log level for the chef run
chef_output_format: Allows to set the format of the chef output

This also provides a fix for #66

### CODE REVIEW

**Comment by Itxaka** [comment]
Tested this with a chef-client version 10.

Also tested with chef-client version 10 + binary pointing to chef-solo (chef-client v10 does not have the -z flag)

File: lib/kitchen/provisioner/dokken.rb [hunk 1]
```diff
@@ -31,6 +31,10 @@ class Dokken < Kitchen::Provisioner::ChefZero
       plugin_version Kitchen::VERSION
 
       default_config :root_path, '/opt/kitchen'
+      default_config :chef_binary, '/opt/chef/embedded/bin/chef-client'
```
Comment (someara) [inline lib/kitchen/provisioner/dokken.rb]: Didn't you just send a PR to change this? ;)

Comment (Itxaka) [inline lib/kitchen/provisioner/dokken.rb]: Yes! You are too fast for me ;)

Comment (Itxaka) [inline lib/kitchen/provisioner/dokken.rb]: Was thinking of leaving it like this, this way the "old" way does not change but peoiple can change it in their configs instead, makes more sense to me. Dont fix what is not broken....

Comment (someara) [inline lib/kitchen/provisioner/dokken.rb]: ¯\_(ツ)_/¯

**Comment by Itxaka** [comment]
Wow that was fast, thanks @someara for the fast response, you just unblock me :+1:

**Comment by someara** [comment]
released in 2.0.14

**Pull request merged**